### PR TITLE
SHO-26-FIX: Social Media Buttons

### PR DESCRIPTION
**Pull request by kevinochoag**
## Describe your changes

This PR adds a new three components. InstagramButton, FacebookButton and TwitterButton:.

## Task URL (Asana, Jira, etc...)

[SHO-26][(https://cultoperrocafe.nifty.pm/l/7nZ8WlIDh3BHT](https://cultoperrocafe.nifty.pm/l/w0ZmsFzoBX2F))

## What was done in this pull request

- [x] Added new component: `<InstagramButton />`.
- [x] Added new component: `<FacebookButton />`.
- [x] Added new component: `<TwitterButton />`.

## Demo / Screenshot / Video
sho-26

### CODE REVIEW

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+export const SocialMediaButton: FC<ButtonProps> = ({
+  socialMediaIcon,
+  ...props
+}) => {
+  let icon;
+  switch (socialMediaIcon) {
+    case 'Instagram':
+      icon = (
+        <a
+          href="https://www.instagram.com/cultoperrocafe/"
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: no need to hardcode this

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          target="_blank"
+          rel="noreferrer"
+        >
+          <div className="socialMedia">
+            <svg
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: have you tried creating a component per SVG and re-use it here?

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+              </g>
+            </svg>
+          </div>
+        </a>
+      );
+      break;
+    case 'Twitter':
+      icon = (
+        <a
+          href="https://twitter.com/cultoperrocafe"
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: no need to hardcode

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+              </g>
+            </svg>
+          </div>
+        </a>
+      );
+      break;
+    case 'Facebook':
+      icon = (
+        <a
+          href="https://www.facebook.com/cultoperrocafe"
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: no need to hardcode

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
+import {FC} from 'react';
+
+import {ButtonProps, StyledButton} from './social-media-button.styles';
+
+export const SocialMediaButton: FC<ButtonProps> = ({
+  socialMediaIcon,
+  ...props
+}) => {
+  let icon;
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: let's try to not use `let`.

**Review by jmsalcido — CHANGES_REQUESTED** [review]

**Comment by jmsalcido** [comment]
instead of having everything in a single component, can you make them a separate component per social platform?

```typescript
<InstagramButton />
<TwitterButton />
<FacebookButton />
<WhateverButton />
```

And that way we can use it differently and add more or remove them depending on what we want to add over the `Footer` and over the ... `Hero` banners or whichever component we want to use.

The way you are thinking this is:

```typescript
<SocialMediaButton socialMediaIcon="twitter"/>
<SocialMediaButton socialMediaIcon="facebook"/>
<SocialMediaButton socialMediaIcon="instagram"/>
```

So... it's super weird to use as an atom too, think it more like `SocialMediaButton icon="" url=""` instead of repeating big names

then you can wrap that over a component named TwitterButton/FacebookButton/WhateverButton.

```typescript
<TwitterButton username="cultoperrocafe" />
<FacebookButton username="cultoperrocafe" />
<InstagramButton username="cultoperrocafe" />
```

(we could make username to be defaulted to `cultoperrocafe` btw... or use a src/href or another thing)

image

image

We would still have 2 components or 1 with the "theme" so... I think it's better to have one component per button, if we could somehow use svg components easily like this: https://blog.logrocket.com/how-to-use-svgs-react/#using-svg-element

Or another way, but.. it makes more sense in my head to have one component per svg... 


Anyway, cc @tonymtz for help :)

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import facebook from './img/facebook.svg';
+import instagramDark from './img/instagram-dark.svg';
+import instagram from './img/instagram.svg';
+import twitterDark from './img/twitter-dark.svg';
+import twitter from './img/twitter.svg';
+import {Icon, StyledButton, Container} from './social-media-button.styles';
+
+interface Props {
+  dark?: boolean;
+}
```
Comment (tonymtz) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: DRY

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+export const InstagramButton: FC<Props> = ({dark = false}) => {
```
Comment (tonymtz) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: Booleans are falsy by default (undefined). You don't need this fallback, you need to adjust your typing :)

File: app/components/atoms/social-media-buttons/social-media-button.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  FacebookButton,
+  TwitterButton,
+} from './social-media-button.component';
+
+export default {
+  title: 'Atoms/SocialMediaButton',
+  component: InstagramButton,
+} as ComponentMeta<typeof InstagramButton>;
+
+const Template: ComponentStory<typeof InstagramButton> = () => (
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.stories.tsx]: same suggestions from https://github.com/tepachelabs/perro-cafe-storefront/pull/33 apply for this.

**Review by jmsalcido — CHANGES_REQUESTED** [review]

**Review by imariel2d — DISMISSED** [review]
LGTM

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+import twitterDark from './img/twitter-dark.svg';
+import twitter from './img/twitter.svg';
+import {
+  Icon,
+  StyledButton,
+  Container,
+  Props,
+} from './social-media-button.styles';
+
+export const InstagramButton: FC<Props> = ({dark}) => {
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: please use props as:

```typescript
interface Props {
  dark?: boolean;
}
```

and keep them in the same file.

```typescript
export const InstagramButton: FC<Props> = (props) => {
    return()
}
```

File: app/components/atoms/social-media-buttons/social-media-button.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+>;
+
+const Template: ComponentStory<
+  typeof InstagramButton | typeof FacebookButton | typeof TwitterButton
+> = (args) => (
+  <>
+    <div
+      style={{display: 'flex', justifyContent: 'space-evenly', width: '20em'}}
+    >
+      <InstagramButton {...args} />
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.stories.tsx]: why do we have a style?, can you please just make a different component per each of those, see: https://github.com/tepachelabs/perro-cafe-storefront/pull/33/files

File: app/components/atoms/social-media-buttons/social-media-button.styles.tsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import {ReactNode} from 'react';
+import styled from 'styled-components';
+
+export interface Props {
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.styles.tsx]: 👎🏽

**Review by jmsalcido — CHANGES_REQUESTED** [review]

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+import facebookDark from './img/facebook-dark.svg';
+import facebook from './img/facebook.svg';
+import instagramDark from './img/instagram-dark.svg';
+import instagram from './img/instagram.svg';
+import twitterDark from './img/twitter-dark.svg';
+import twitter from './img/twitter.svg';
+import {Icon, StyledButton, Container} from './social-media-button.styles';
+
+export default interface Props {
```
Comment (jmsalcido) [inline app/components/atoms/social-media-buttons/social-media-button.component.tsx]: no need to export

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx (hunk 9 above)
Comment (jmsalcido) [reply app/components/atoms/social-media-buttons/social-media-button.component.tsx]: what about this?

**Review by jmsalcido — CHANGES_REQUESTED** [review]
image

something seems off... the "dark" set to true confuses me a lot lol, is it just me? @imariel2d  @tonymtz

File: app/components/atoms/social-media-buttons/social-media-button.component.tsx (hunk 12 above)
Comment (kevinochoag) [reply app/components/atoms/social-media-buttons/social-media-button.component.tsx]: image
**But I use this export to import into the Styles File** 
image

**Comment by imariel2d** [comment]
@jmsalcido agree, seems its flipped.

**Review by jmsalcido — CHANGES_REQUESTED** [review]
https://github.com/tepachelabs/perro-cafe-storefront/pull/31#discussion_r1159166010

Not addressed and props still on different file

**Comment by jmsalcido** [comment]
> > [#31 (comment)](https://github.com/tepachelabs/perro-cafe-storefront/pull/31#discussion_r1159166010)
> > Not addressed and props still on different file
> 
> The interface is required to be in the styles file, otherwise it will cause a circular dependency :)

expecting reply from @Kevstrosky

**Review by jmsalcido — APPROVED** [review]

**Pull request merged**